presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java

```diff
@@ -451,7 +453,28 @@ public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle
             dao.updateTableVersion(table.getTableId(), session.getStartTime());
         });
 
-        shardManager.addColumn(table.getTableId(), new ColumnInfo(columnId, column.getType()));
+        addShardColumnWithRetry(table, new ColumnInfo(columnId, column.getType()));
+    }
+
+    private void addShardColumnWithRetry(RaptorTableHandle table, ColumnInfo newColumn)
+    {
+        int attempts = 0;
+        boolean ok;
+        do {
+            attempts++;
+            try {
+                shardManager.addColumn(table.getTableId(), newColumn);
+                ok = true;
```

Comment:
There's no need for this variable. You can loop unconditionally and return here.